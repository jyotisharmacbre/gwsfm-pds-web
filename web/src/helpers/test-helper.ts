import checkPropTypes from 'check-prop-types';

export const findByTestAtrr = (component: any, attr: string) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component: any, expectedProps: any) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    'props',
    component.name
  );
  return propsErr;
};
