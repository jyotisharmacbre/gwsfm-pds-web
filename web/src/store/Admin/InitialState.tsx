import { IAdminState } from './Types/IAdminState';
import { IAdminDefaults } from './Types/IAdminDefault';

export const initialState: IAdminState = {
	adminDefaultValues: [],
	error: ''
};
