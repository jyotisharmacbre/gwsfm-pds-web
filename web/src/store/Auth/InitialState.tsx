import { IAuth } from "./Types/IAuth";

export const initialState: IAuth= {
    token: null,
    name: null,
    firstName: null,
    email: null
};
