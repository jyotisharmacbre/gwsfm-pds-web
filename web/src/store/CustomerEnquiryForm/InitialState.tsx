import { IProjectDetailState } from './Types/IProjectDetailState';
import EventType from '../../enums/EventType';
import Notify from '../../enums/Notify';

export const initialState: IProjectDetailState = {
    form: {
        projectId: '',
        name: '',
        contractorId: '',
        companyId: '',
        headOfProject: '',
        projectOwner: '',
        projectManager: '',
        pmHasExperience: false,
        scope: '',
        cnNumber: '',
        status: 1,
        engagementId: undefined,
        countryId: 0,
        currencyId: 0,
        probabilityOfWinning: '',
        approxValue: '',
        contractTypeId: '',
        cdmNotifiable: false,
        firstAssetWorkedOn: undefined,
        secondAssetWorkedOn: undefined,
        thirdAssetWorkedOn: undefined,
        comment: '',
        otherCompanyName: '',
        otherContractName: '',
        divisionId: '',
        otherDivision: '',
        businessUnitId: '',
        otherBusinessUnit: '',
        weightedTCV: undefined,
        soldMargin: undefined,
        otherEngagementType: '',
        otherFirstAssetWorkedOn: '',
        otherSecondAssetWorkedOn: '',
        otherThirdAssetWorkedOn: ''
    },
    enquiryOverview: {
        projectName: '',
        contractorId: '',
        headOfProject: '',
        projectManager: '',
        scope: '',
        cnNumber: '',
        otherContractName: ''
    },
    error: null,
    loading: false,
    notify: Notify.none,
    event: EventType.none,
    enquiryOverviewError: null
};