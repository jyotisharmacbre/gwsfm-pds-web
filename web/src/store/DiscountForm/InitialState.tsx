import { IDiscountState } from './Types/IDiscountState';
import Notify from '../../enums/Notify';
import EventType from '../../enums/EventType';
import IDiscountSubContractor from './Types/IDiscountSubContractor';

export const newSubContractorDiscount: IDiscountSubContractor = {
  subContractorDiscountId: '',
  projectId: '',
  supplierName: '',
  supplierState: '',
  supplierTotalDiscount: undefined,
  supplierComments: '',
}

export const initialState: IDiscountState = {
  form: {
    projectId: '',
    clientDiscount: {
      discountId: '',
      projectId: '',
      discount: undefined,
      clientState: '',
      discountType: 1,
      clientComments: ''
    },
    subContractorDiscounts: [{ ...newSubContractorDiscount }],
  },
  error: null,
  loading: false,
  notify: Notify.none,
  event: EventType.none
};