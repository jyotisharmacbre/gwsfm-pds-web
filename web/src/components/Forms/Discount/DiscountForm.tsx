import React, { Component } from 'react';
import {
  Field,
  reduxForm,
  InjectedFormProps,
  formValueSelector
} from 'redux-form';
import PdsFormInput from '../../PdsFormHandlers/PdsFormInput';
import PdsFormSelect from '../../PdsFormHandlers/PdsFormSelect';
import PdsFormTextArea from '../../PdsFormHandlers/PdsFormTextArea';
import PdsFormButton from '../../PdsFormHandlers/PdsFormButton';
import { selectionButtons } from '../../../helpers/constants';
import { Validate, alphaNumeric, onlyNumber } from '../../../helpers/fieldValidations';
import { connect } from 'react-redux';
import { IState } from '../../../store/state';
import { FormattedMessage, injectIntl } from 'react-intl';
import { LookupType } from '../../../store/Lookups/Types/LookupType';
import { getDropdown } from '../../../helpers/utility-helper';
import PdsFormTypeAhead from '../../PdsFormHandlers/PdsFormTypeAhead';
import { IProjectDetail } from '../../../store/CustomerEnquiryForm/Types/IProjectDetail';
import { ICurrency } from '../../../store/Lookups/Types/ICurrency';
import IReactIntl from '../../../Translations/IReactIntl';

interface Props{

}

const DiscountForm: React.FC<
  Props & IReactIntl & InjectedFormProps<any>
> = (props: any) => {
  
    return (
      <div className="row">
        <div className="col-lg-12">
          <div className="Discountforms_wrap">
            <div className="row">
              <div className="col-lg-7">
                <form className="custom-wrap p-0">
                  <h2><FormattedMessage id="TITLE_SUBCONTRACTOR_DISCOUNT"></FormattedMessage></h2>
                    <Field
                      name="name"
                      type="text"
                      component={PdsFormInput}
                      className="required"
                      validate={[
                        Validate.required('LABEL_SUPPLIER'),
                        Validate.maxLength(1000)
                      ]}
                      messageKey="MESSAGE_SUPPLIER_NAME"
                      labelKey="LABEL_SUPPLIER"
                      placeholderKey="PLACEHOLDER_ENTER_SUPPLIER_NAME"
                    />
                    <Field
                      name="name"
                      type="text"
                      component={PdsFormInput}
                      className="required"
                      validate={[
                        Validate.required('LABEL_STATE_DETAILS'),
                        Validate.maxLength(1000)
                      ]}
                      messageKey="MESSAGE_STATE_DETAILS_NAME"
                      labelKey="LABEL_STATE_DETAILS"
                      placeholderKey="PLACEHOLDER_ENTER_STATE_DETAILS"
                    /> 
                  <div className="form-group">
                    <label><FormattedMessage id="LABEL_TOTAL_DISCOUNT"></FormattedMessage></label>
                    <input
                      className="width-120 form-control"
                      type="text"
                      placeholder=""
                    />
                    <span className="symbol_fix">&#163;</span>
                    <label className="w-100 mb-0 mt-3">
                    <FormattedMessage id="LABEL_SUB_TOTAL_DISCOUNTS"></FormattedMessage> 
                    </label>
                    <label className="m-0"><FormattedMessage id="LABEL_200000"></FormattedMessage></label>
                    </div>
                  <Field
                  name="scope"
                  rows="6"
                  component={PdsFormTextArea}
                  className="required"
                  validate={[
                    Validate.required('LABEL_COMMENTS'),
                    Validate.maxLength(1040)
                  ]}             
                  labelKey="LABEL_COMMENTS"
                  placeholderKey="PLACEHOLDER_TYPE_IN_ANY_ADDITIONAL_COMMENTS"
                />
                </form>
              </div>
            </div>

            <div className="hr_line"></div>

            <div className="row">
              <div className="col-lg-7">
                <form className="custom-wrap p-0">
                  <h2><FormattedMessage id="TITAL_CLIENT_DISCOUNT"></FormattedMessage></h2>
                  <Field
                      name="name"
                      type="text"
                      component={PdsFormInput}
                      className="required"
                      validate={[
                        Validate.required('LABEL_CLIENT'),
                        Validate.maxLength(1000)
                      ]}
                      messageKey="MESSAGE_CLIENT_NAME"
                      labelKey="LABEL_CLIENT"
                      placeholderKey="PLACEHOLDER_ENTER_CLIENT_NAME"
                    />
                  <Field
                      name="name"
                      type="text"
                      component={PdsFormInput}
                      className="required"
                      validate={[
                        Validate.required('STATE_DETAILS'),
                        Validate.maxLength(1000)
                      ]}
                      messageKey="MESSAGE_STATE_DETAILS"
                      labelKey="LABEL_STATE_DETAILS"
                      placeholderKey="PLACEHOLDER_ENTER_STATE_DETAILS"
                    />
                  <div className="form-group">
                    <label><FormattedMessage id="LABEL_DISCOUNT_TYPE"></FormattedMessage></label>
                    <div className="form-check">
                      <input
                        type="radio"
                        className="form-check-input"
                        name="Value"
                        id="1"
                      />
                      <label className="form-check-label" htmlFor="1">
                      <FormattedMessage id="LABEL_PERCENTAGE"></FormattedMessage>
                      </label>
                    </div>
                    <div className="form-check">
                      <input
                        type="radio"
                        className="form-check-input"
                        name="Value"
                        id="2"
                      />
                      <label className="form-check-label" htmlFor="2">
                        LABEL_VALUE
                      </label>
                    </div>
                  </div>
                  <div className="form-group">
                    <label>LABEL_DISCOUNT</label>
                    <input
                      className="width-250 form-control"
                      type="text"
                      placeholder=""
                    />
                    <span className="symbol_fix">PLACEHOLDER_QUOTE_SUPPLIER2</span>
                    <label className="w-100 mb-0 mt-3">
                      LABEL_SUB_TOTAL_DISCOUNTS
                    </label>
                    <label className="m-0">LABEL_&#163;2,000.00</label>
                  </div>
                  <Field
                  name="scope"
                  rows="6"
                  component={PdsFormTextArea}
                  className="required"
                  validate={[
                    Validate.required('LABEL_COMMENTS'),
                    Validate.maxLength(1040)
                  ]}             
                  labelKey="LABEL_COMMENTS"
                  placeholderKey="PLACEHOLDER_TYPE_IN_ANY_ADDITIONAL_COMMENTS"
                />
                </form>
              </div>
            </div> 

            <div className="hr_line mb-0 mt-4"></div>

            <div className="mr-35 three-btn">
              <button className="active" type="button">
                BUTTON_PREVIOUS
              </button>
              <button type="button" name="next" className="active ml-auto">
                BUTTON_SAVE
              </button>
              <button type="button" name="next">
                BUTTON_NEXT
              </button>
            </div>
          </div>
        </div>
      </div>
    );
  }

  const form = reduxForm<IProjectDetail, Props>({
    form: 'DiscountForm',
    enableReinitialize: true
  })(injectIntl(DiscountForm));
  
  export default connect(null)(form);


