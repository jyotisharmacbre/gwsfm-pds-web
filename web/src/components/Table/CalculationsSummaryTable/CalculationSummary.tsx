import { calculateSell, calculateAverageMargin,calculateClientDiscount} from '../../../helpers/formulas';
import {ISubContractorActivity} from '../../../store/SubContractor/Types/ISubContractorActivity';
import ISummaryCalculation from '../../../store/SummaryCalculation/Types/ISummaryCalculation';
import {IPreliminariesComponentDetails} from '../../../store/Preliminaries/Types/IPreliminariesComponentDetails';
import {IPreliminariesItems} from '../../../store/Preliminaries/Types/IPreliminariesItems';
import {IDiscountActivity} from '../../../store/DiscountForm/Types/IDiscountActivity';

export const getPreliminarySummaryCalculation = (data:Array<IPreliminariesComponentDetails>,state:ISummaryCalculation) => {
        if(data.length >0)
        {
          data.map((details:IPreliminariesComponentDetails)=>{
            details.items.map((element:IPreliminariesItems)=>{
            state.cost = state.cost + (+element.totalCost);
            state.sell = state.sell + (+calculateSell(element.totalCost,element.grossMargin));
            state.margin = state.margin + (+element.grossMargin);
          })
          })
        }
        return {cost:state.cost,sell:state.sell,margin:state.margin};  
}

export const getDiscountSummaryCalculation = (data:IDiscountActivity,state:ISummaryCalculation) => {
    if(data.supplierTotalDiscount != undefined)
      state.cost = state.cost - data.supplierTotalDiscount;
    if(data.clientDiscount != undefined){
      state.sell = state.sell - (+calculateClientDiscount(data.discountType,state.sell,data.clientDiscount));
    }
    state.margin = +calculateAverageMargin(state.cost,state.sell)
    return {cost:state.cost,sell:state.sell,margin:state.margin};
}

export const getSubContractorSummaryCalculation = (data:Array<ISubContractorActivity>,state:ISummaryCalculation) => {
        data.map((element:ISubContractorActivity)=>{
            state.cost = state.cost + (+element.totalCost);
            state.sell = state.sell + (+calculateSell(element.totalCost,element.grossMargin));
            state.margin = state.margin + (+element.grossMargin);
        })
    return {cost:state.cost,sell:state.sell,margin:state.margin};
}