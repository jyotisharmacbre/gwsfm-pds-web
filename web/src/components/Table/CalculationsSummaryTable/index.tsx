import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { IState } from '../../../store/state';
import * as actions from '../../../store/rootActions';
import { ISubContractorActivity } from '../../../store/SubContractor/Types/ISubContractorActivity';
import { calculateTotalSum, calculateSell, calculateAverageMargin } from '../../../helpers/formulas';
import {
	getSubContractorSummaryCalculation,
	getPreliminarySummaryCalculation,
	getDiscountSummaryCalculation
} from '../../../helpers/pricing-calculation-helper';
import ISummaryCalculation from '../../../models/ISummaryCalculation';
import IPricing from '../../../models/IPricing';
import { IPreliminariesComponentDetails } from '../../../store/Preliminaries/Types/IPreliminariesComponentDetails';
import { FormattedMessage } from 'react-intl';
import { IDiscountActivity } from '../../../store/DiscountForm/Types/IDiscountActivity';
import { ICurrency } from '../../../store/Lookups/Types/ICurrency';
import Currency from '../../../store/Lookups/InitialState/Currency';
import { getPropertyName, getFilterElementFromArray } from '../../../helpers/utility-helper';

interface Props {
	subContractor?: Array<ISubContractorActivity>;
	preliminary?: Array<IPreliminariesComponentDetails>;
	discount?: {} | IDiscountActivity;
	currencySymbol: string;
}

const CalculationsSummaryTable: React.FC<Props> = (props) => {
	let initPricing: IPricing = { cost: 0, sell: 0, margin: 0 };
	let initSummary: ISummaryCalculation = { cost: 0, sell: 0, margin: 0, grossMargin: 0 };
	const [ summaryCalculation, setSummaryCalculationState ] = useState<ISummaryCalculation>({ ...initSummary });
	const [ subContractorData, setSubContractorData ] = useState({ ...initPricing });
	const [ preliminaryData, setPreliminaryData ] = useState({ ...initPricing });
	useEffect(
		() => {
			if (props.subContractor && props.subContractor[0].projectId) {
				setSubContractorData(getSubContractorSummaryCalculation(props.subContractor));
			}
		},
		[ props.subContractor ]
	);

	useEffect(
		() => {
			if (props.preliminary && props.preliminary.length > 0) {
				setPreliminaryData(getPreliminarySummaryCalculation(props.preliminary));
			}
		},
		[ props.preliminary ]
	);

	useEffect(
		() => {
			if (props.discount && subContractorData && preliminaryData) {
				setSummaryCalculationState(
					getDiscountSummaryCalculation(
						props.discount as IDiscountActivity,
						subContractorData,
						preliminaryData
					)
				);
			}
		},
		[ props.discount, subContractorData, preliminaryData ]
	);
	return (
		<div className="col-lg-12 px-0">
			<div className="price-sumry discount_table">
				<div className="inner-block">
					<table className="price-table">
						<thead>
							<tr>
								<th>
									<FormattedMessage id="T_HEADING_TOTAL_COST" />{' '}
								</th>
								<th>
									<FormattedMessage id="T_HEADING_TOTAL_MARGIN" />{' '}
								</th>
								<th>
									<FormattedMessage id="T_HEADING_GROSS_MARGIN" />{' '}
								</th>
								<th>
									<FormattedMessage id="T_HEADING_TOTAL_SELL" />{' '}
								</th>
							</tr>
						</thead>
						<tbody>
							<tr>
								<td>
									{props.currencySymbol}
									<span data-test="total-cost-summary">{summaryCalculation.cost}</span>
								</td>
								<td>
									<span data-test="total-margin-summary">{summaryCalculation.margin}</span>
									(%)
								</td>
								<td>
									{props.currencySymbol}
									<span data-test="gross-margin-summary">{summaryCalculation.grossMargin}</span>
								</td>
								<td>
									{props.currencySymbol}
									<span data-test="total-sell-summary">{summaryCalculation.sell.toFixed(2)}</span>
								</td>
							</tr>
						</tbody>
					</table>
				</div>
			</div>
		</div>
	);
};

export default CalculationsSummaryTable;
