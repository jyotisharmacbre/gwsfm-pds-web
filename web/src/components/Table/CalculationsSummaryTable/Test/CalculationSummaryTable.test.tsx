import React from 'react';
import { mount } from 'enzyme';
import { applyMiddleware, createStore, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import translations from '../../../../Translations/translation';
import CalculationsSummaryTable from '../index';
import { initialState as subContractorInitialState, newActivity } from '../../../../store/SubContractor/InitialState';
import { initialState as preliminariesInitialState } from '../../../../store/Preliminaries/InitialState';
import { findByTestAtrr, checkProps } from '../../../../helpers/test-helper';
import { initialState as discountInitialState } from '../../../../store/DiscountForm/InitialState';

let wrapper: any;

const mountCalculationSummaryTable = (props) => {
	wrapper = mount(
		<IntlProvider locale="en" messages={translations['en'].messages}>
			<CalculationsSummaryTable {...props} />
		</IntlProvider>
	);
};

describe('should calculation summary component renders without error', () => {
	let Props = {
		preliminary: preliminariesInitialState,
		subContractor: subContractorInitialState.form.activities,
		discount: discountInitialState,
		currencySymbol: '$'
	};
	Props.subContractor[0].projectId = 'test';
	beforeEach(() => {
		mountCalculationSummaryTable(Props);
	});
	it('defines the component', () => {
		expect(wrapper).toBeDefined();
	});

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should Not throw a warning for proptypes', () => {
		const expectedProps = {
			preliminary: preliminariesInitialState,
			subContractor: subContractorInitialState.form.activities,
			discount: discountInitialState,
			currencySymbol: '$'
		};
		const propsError = checkProps(CalculationsSummaryTable, expectedProps);
		expect(propsError).toBeUndefined();
	});
});
describe('should calculation summary component, calculate the cost, margin and sell correctly', () => {
	let Props = {
		preliminary: preliminariesInitialState,
		subContractor: subContractorInitialState.form.activities,
		discount: discountInitialState.form,
		currencySymbol: '$'
	};
	it('should calculate the pricing summary correctly, if discount is Zero', () => {
		Props.subContractor[0].totalCost = 100;
		Props.subContractor[0].grossMargin = 20;
		Props.discount.clientDiscount = 0;
		Props.discount.supplierTotalDiscount = 0;
		mountCalculationSummaryTable(Props);
		expect(findByTestAtrr(wrapper, 'total-cost-summary').text()).toEqual('100.00');
		expect(findByTestAtrr(wrapper, 'total-margin-summary').text()).toEqual('20');
		expect(findByTestAtrr(wrapper, 'gross-margin-summary').text()).toEqual('25');
		expect(findByTestAtrr(wrapper, 'total-sell-summary').text()).toEqual('125.00');
	});

	it('should calculate the pricing summary correctly if only sub contractor discount is applying', () => {
		Props.subContractor[0].totalCost = 100;
		Props.subContractor[0].grossMargin = 20;
		Props.discount.clientDiscount = 0;
		Props.discount.supplierTotalDiscount = 10;
		mountCalculationSummaryTable(Props);
		expect(findByTestAtrr(wrapper, 'total-cost-summary').text()).toEqual('90.00');
		expect(findByTestAtrr(wrapper, 'total-margin-summary').text()).toEqual('28');
		expect(findByTestAtrr(wrapper, 'gross-margin-summary').text()).toEqual('35');
		expect(findByTestAtrr(wrapper, 'total-sell-summary').text()).toEqual('125.00');
	});

	it('should calculate the pricing summary correctly if only client discount(%) is applying', () => {
		Props.subContractor[0].totalCost = 100;
		Props.subContractor[0].grossMargin = 20;
		Props.discount.discountType = 1;
		Props.discount.clientDiscount = 10;
		Props.discount.supplierTotalDiscount = 0;
		mountCalculationSummaryTable(Props);
		expect(findByTestAtrr(wrapper, 'total-cost-summary').text()).toEqual('100.00');
		expect(findByTestAtrr(wrapper, 'total-margin-summary').text()).toEqual('11.11');
		expect(findByTestAtrr(wrapper, 'gross-margin-summary').text()).toEqual('12.5');
		expect(findByTestAtrr(wrapper, 'total-sell-summary').text()).toEqual('112.50');
	});

	it('should calculate the pricing summary correctly if only client discount(value) is applying', () => {
		Props.subContractor[0].totalCost = 100;
		Props.subContractor[0].grossMargin = 20;
		Props.discount.discountType = 2;
		Props.discount.clientDiscount = 10;
		Props.discount.supplierTotalDiscount = 0;
		mountCalculationSummaryTable(Props);
		expect(findByTestAtrr(wrapper, 'total-cost-summary').text()).toEqual('100.00');
		expect(findByTestAtrr(wrapper, 'total-margin-summary').text()).toEqual('13.04');
		expect(findByTestAtrr(wrapper, 'gross-margin-summary').text()).toEqual('15');
		expect(findByTestAtrr(wrapper, 'total-sell-summary').text()).toEqual('115.00');
	});

	it('should calculate the pricing summary correctly after applying discount in percentage', () => {
		Props.subContractor[0].totalCost = 100;
		Props.subContractor[0].grossMargin = 20;
		Props.discount.discountType = 1;
		Props.discount.clientDiscount = 10;
		Props.discount.supplierTotalDiscount = 10;
		mountCalculationSummaryTable(Props);
		expect(findByTestAtrr(wrapper, 'total-cost-summary').text()).toEqual('90.00');
		expect(findByTestAtrr(wrapper, 'total-margin-summary').text()).toEqual('20');
		expect(findByTestAtrr(wrapper, 'gross-margin-summary').text()).toEqual('22.5');
		expect(findByTestAtrr(wrapper, 'total-sell-summary').text()).toEqual('112.50');
	});

	it('should calculate the pricing summary correctly after applying discount in value', () => {
		Props.subContractor[0].totalCost = 100;
		Props.subContractor[0].grossMargin = 20;
		Props.discount.discountType = 2;
		Props.discount.clientDiscount = 10;
		Props.discount.supplierTotalDiscount = 10;
		mountCalculationSummaryTable(Props);
		expect(findByTestAtrr(wrapper, 'total-cost-summary').text()).toEqual('90.00');
		expect(findByTestAtrr(wrapper, 'total-margin-summary').text()).toEqual('21.74');
		expect(findByTestAtrr(wrapper, 'gross-margin-summary').text()).toEqual('25');
		expect(findByTestAtrr(wrapper, 'total-sell-summary').text()).toEqual('115.00');
	});

	it('should calculate the pricing summary correctly after applying discount in percentage on more than 1 activity', () => {
		Props.subContractor[0].totalCost = 100;
		Props.subContractor[0].grossMargin = 20;
		Props.subContractor.push({ ...newActivity });
		Props.subContractor[1].totalCost = 100;
		Props.subContractor[1].grossMargin = 20;
		Props.discount.discountType = 1;
		Props.discount.clientDiscount = 10;
		Props.discount.supplierTotalDiscount = 10;
		mountCalculationSummaryTable(Props);
		expect(findByTestAtrr(wrapper, 'total-cost-summary').text()).toEqual('190.00');
		expect(findByTestAtrr(wrapper, 'total-margin-summary').text()).toEqual('15.56');
		expect(findByTestAtrr(wrapper, 'gross-margin-summary').text()).toEqual('35');
		expect(findByTestAtrr(wrapper, 'total-sell-summary').text()).toEqual('225.00');
	});

	it('should calculate the pricing summary correctly after applying discount in value on more than 1 activity', () => {
		Props.subContractor[0].totalCost = 100;
		Props.subContractor[0].grossMargin = 20;
		Props.subContractor.push({ ...newActivity });
		Props.subContractor[1].totalCost = 100;
		Props.subContractor[1].grossMargin = 20;
		Props.discount.discountType = 2;
		Props.discount.clientDiscount = 10;
		Props.discount.supplierTotalDiscount = 10;
		mountCalculationSummaryTable(Props);
		expect(findByTestAtrr(wrapper, 'total-cost-summary').text()).toEqual('190.00');
		expect(findByTestAtrr(wrapper, 'total-margin-summary').text()).toEqual('20.83');
		expect(findByTestAtrr(wrapper, 'gross-margin-summary').text()).toEqual('50');
		expect(findByTestAtrr(wrapper, 'total-sell-summary').text()).toEqual('240.00');
	});
});
