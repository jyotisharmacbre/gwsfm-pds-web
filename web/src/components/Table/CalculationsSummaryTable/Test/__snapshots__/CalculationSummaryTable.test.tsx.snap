// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculation summary component tests should match the snapshot 1`] = `undefined`;
